platform/engineering/design-docs/2020-09-24-package-forms-library.md

```diff
@@ -0,0 +1,217 @@
+- [`FormSaved.jsx`](https://github.com/department-of-veterans-affairs/vets-website/blob/cc4a3172edc242bf60c93aec5b170c734dc57985/src/platform/forms/save-in-progress/FormSaved.jsx#L10)
+- [`AuthorizationComponent.jsx`](https://github.com/department-of-veterans-affairs/vets-website/blob/cc4a3172edc242bf60c93aec5b170c734dc57985/src/platform/forms/components/AuthorizationComponent.jsx#L12)
+
+Finally, there are some testing helpers that live in the `src/applications/hca` directory and are used in forms library tests:
+
+- [`01-sip-autosave.e2e.spec.js`](https://github.com/department-of-veterans-affairs/vets-website/blob/cc4a3172edc242bf60c93aec5b170c734dc57985/src/platform/forms/tests/save-in-progress/01-sip-autosave.e2e.spec.js#L4)
+- [`01-sip-fail-load.e2e.spec.js`](https://github.com/department-of-veterans-affairs/vets-website/blob/cc4a3172edc242bf60c93aec5b170c734dc57985/src/platform/forms/tests/save-in-progress/01-sip-load-fail.e2e.spec.js#L3)
+- [`01-sip-finish-later.e2e.spec.js`](https://github.com/department-of-veterans-affairs/vets-website/blob/cc4a3172edc242bf60c93aec5b170c734dc57985/src/platform/forms/tests/save-in-progress/01-sip-finish-later.e2e.spec.js#L4)
+- [`01-sip-review.e2e.spec.js`](https://github.com/department-of-veterans-affairs/vets-website/blob/cc4a3172edc242bf60c93aec5b170c734dc57985/src/platform/forms/tests/save-in-progress/01-sip-review.e2e.spec.js#L3)
+
+
+#### Platform entanglement
+
+In addition to the forms library, the platform provides other code that is used both in the forms library and in applications. 
+
+- `utilities`
+- `monitoring`
+- `site-wide`
+- `user`
+- `static-data`
+
+These aren't _strictly_ part of the forms library, but since the forms library makes use of its parent platform code, it becomes more difficult to place clear boundaries around the forms library.
+
+### High Level Design
+
+The forms library will be consolidated into a single place and bundled as a Node package.
+This will allow VFS teams to import it as if it were any other Node module.
+
+This change should not have a noticeable functional impact on individual app teams.
+Towards the end the only change that should be made on the app side of things is changing where forms library code/styles are imported from.
+
+<!-- The source for this diagram is the `*.odg` file in the same directory, a LibreOffice Draw doc-->
+![Forms library bundling](images/forms-library-package/forms-library-package.svg)
+
+## Specifics
+
+### Detailed Design
+
+A few things will have to happen in order pave the way for additional changes that will bundle the forms library.
+
+#### Preparation
+
+- The app-specific code mentioned in [Application Entanglement](#application-entanglement) will have to be moved so that it no longer lives in the forms library.
+- Use the `no-unresolved-modules` ESLint plugin to [find & remove unused code](https://github.com/department-of-veterans-affairs/va.gov-team/issues/8763) in the `forms` and `forms-library` directories
```

Comment:
Is this a non-goal? If so, this could definitely go in `Future Work`.
> Refactoring the forms library beyond what is required to group the existing code together in the same package

Reply:
I don't think so - I view it as removing pieces of code which do not need to be migrated into the `forms-library` package.